tests/testSphPoint.py

```diff
@@ -0,0 +1,615 @@
+                continue
+            point1 = SphPoint(lon1*degrees, lat1*degrees)
+            for lon2, lat2 in self._dataset[0:20]:
+                if lon1 == lon2 and lat1 == lat2:
+                    continue
+                if abs(lat2) == 90.0:
+                    continue
+                point2 = SphPoint(lon2*degrees, lat2*degrees)
+                bearing = point1.bearingTo(point2)
+                distance = point1.separationTo(point2)
+
+                newPoint = point1.offset(bearing, distance)
+                self.assertIsInstance(newPoint, SphPoint)
+                self.assertAnglesNearlyEqual(point2.getLongitude(), newPoint.getLongitude())
+                self.assertAnglesNearlyEqual(point2.getLatitude(), newPoint.getLatitude())
+
+    def testIterResult(self):
+        """Test if iteration returns the expected values.
+        """
+        for lon, lat in self._dataset:
+            point = SphPoint(lon*degrees, lat*degrees)
+            # Test mechanics directly
+            it = iter(point)
+            self.assertEqual(point.getLongitude(), next(it))
+            self.assertEqual(point.getLatitude(), next(it))
+            with self.assertRaises(StopIteration):
+                next(it)
+
+            # Intended use case
+            lon, lat = point
+            self.assertEqual(point.getLongitude(), lon)
+            self.assertEqual(point.getLatitude(), lat)
+
+    def testStrValue(self):
+        """Test if __str__ produces output consistent with its spec.
+
+        This is necessarily a loose test, as the behavior of __str__
+        is (deliberately) incompletely specified.
+        """
+        for lon, lat in self._dataset:
+            point = SphPoint(lon*degrees, lat*degrees)
+            numbers = re.findall(r'(?:\+|-)?[\d.]+', str(point))
+            self.assertEqual(2, len(numbers),
+                             "String '%s' should have exactly two coordinates." % (point,))
+
+            # Low precision to allow for only a few digits in string.
+            self.assertAlmostEqual(point.getLongitude().asDegrees(), float(numbers[0]), delta=1e-6)
+            self.assertAlmostEqual(point.getLatitude().asDegrees(), float(numbers[1]), delta=1e-6)
+
+            self.assertTrue(numbers[1].startswith("+") or numbers[1].startswith("-"))
+
+    def testReprValue(self):
+        """Test if __repr__ is a machine-readable representation.
+        """
+        for lon, lat in self._dataset:
+            point = SphPoint(lon*degrees, lat*degrees)
+            pointRepr = repr(point)
+            className = type(point).__name__
+            self.assertIn("degrees", pointRepr)
+            self.assertEqual(2, len(pointRepr.split(",")))
+
+            copy = eval(pointRepr)
+            self.assertAnglesNearlyEqual(point.getLongitude(), copy.getLongitude())
+            self.assertAnglesNearlyEqual(point.getLatitude(),  copy.getLatitude())
+
+    def _assertPointsNearlyEqual(self, lhs, rhs, *args):
```

Comment:
I strongly suggest you test that the separation is less than a given value, instead of comparing longitude and latitude, because the measure is invariant over the sphere.

Please add this to `afw.geom.utils` (using separation as the test). The code will be essentially identical to `assertCoordsNearlyEqual` in lsst.coords.utils`. This will be a big help to anyone using `SphCoord` in tests, as it helps avoid naive errors, such as testing that longitude values are nearly equal when two points may be nearly the same but in different wraps.

I am less convinced `assertPointsNotNearlyEqual` is needed as a publicly available assertion, but it's up to you.

If you do, a test will be required in `tests/testMethods.py`.